lib/protocol/protocol.go

```diff
@@ -322,9 +324,20 @@ func (c *rawConnection) Request(folder string, name string, offset int64, size i
 	return res.val, res.err
 }
 
-// ClusterConfig send the cluster configuration message to the peer and returns any error
+// ClusterConfig sends the cluster configuration message to the peer.
+// It must be called just once (as per BEP).
 func (c *rawConnection) ClusterConfig(config ClusterConfig) {
-	c.send(&config, nil)
+	select {
+	case <-c.sentClusterConfig:
+		return
```

Comment:
Why this? I think it might be fine to send multiple of these in the future.

Reply:
Not unless the BEP gets changed:

> A Cluster Config message MUST be the first post authentication message sent on a BEP connection. Additional Cluster Config messages MUST NOT be sent after the initial exchange.

Also the code as it is would not work at all if multiple calls to `ClusterConfig` were allowed, as it sends outside of the send loop and sends must never happen concurrently. Of course this can be done differently, it's just a bit more complicated and in my opinion unnecessary due to BEP.

Reply:
Fine